### PR TITLE
doc/dev: add formatting to basic workflow

### PR DESCRIPTION
**Pull request by zdover23**
Add some formatting to a few sections near the end of doc/dev/developer_guide_basic-workflow.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/dev/developer_guide/basic-workflow.rst [hunk 1]
```diff
@@ -501,17 +501,17 @@ the merge message input will be automatically populated.
 Using .githubmap to Find a Reviewer's Email Address
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 If you cannot find the email address of the reviewer on his or her GitHub
-page, you can look it up in the **.githubmap** file, which can be found in
-the repository at **/ceph/.githubmap**.
+page, you can look it up in the ``.githubmap`` file, which can be found in
+the repository at ``/ceph/.githubmap``.
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/basic-workflow.rst:505]: I think you want to remove the first slash

File: doc/dev/developer_guide/basic-workflow.rst [hunk 2]
```diff
@@ -521,9 +521,9 @@ Using ptl-tool to Generate Merge Commits
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Another method of generating merge commits involves using Patrick Donnelly's
-**ptl-tool** pull commits. This tool can be found at
-**/ceph/src/script/ptl-tool.py**.  Merge commits that have been generated by
-the **ptl-tool** have the following form::
+``ptl-tool`` to pull commits. This tool can be found at
+``/ceph/src/script/ptl-tool.py``.  Merge commits that have been generated by the
```
Comment (anthonyeleven) [inline doc/dev/developer_guide/basic-workflow.rst:525]: Remove initial slash?

**Review by anthonyeleven — APPROVED** [review]

File: doc/dev/developer_guide/basic-workflow.rst (hunk 1 above)
Comment (zdover23) [reply doc/dev/developer_guide/basic-workflow.rst:505]: I have the opposite intuition, but I'm willing to be argued out of it. I thought that the initial slash here made it clear that the location begins at the root ``/`` of the repository's filesystem.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/58737 - Squid backport
https://github.com/ceph/ceph/pull/58738 - Reef backport
https://github.com/ceph/ceph/pull/58739 - Quincy backport

**Pull request merged**